### ISSUE TITLE
v0.63.0 Discussion

### ISSUE THREAD

**Issue by TheSavior** [react-native-community/releases#186]
## Highlighted Changes
- Native Colors (PlatformColor, DynamicColorIOS)
- LogBox
- Pressable
- Dropping iOS 9 support
- Dropping Node.js 8 support (>=10 is now required)
- [Android Support for rendering Views in Text without explicit size](https://github.com/facebook/react-native/commit/dbb7eacb429adb4160e740017c212bfd6df0f03a)

## Work Required

- [x] Changelog @jeremy-deutsch 
- [x] Blog post @grabbou

<details>
  <summary>Click to expand!</summary>

- [x] facebook/react-native#28776
- [x] Fix [`scrollResponderScrollNativeHandleToKeyboard` warnings](https://github.com/facebook/react-native/issues/28203#issuecomment-592973087). Fix [here](https://github.com/facebook/react-native/commit/3246f68952c8fe065367ef8d2cdf7da5dfff78e4)
- [x] [A new Image Prop `analyticTag` was accidentally landed. This needs to get prefixed as `internal`](https://github.com/facebook/react-native/commit/1c10568967231cce2be01d2341f4edb79d57ec48) @dvacca. Fix [here](https://github.com/facebook/react-native/commit/8e48dc0555e0c3432a77d421b1e535d17f5933c8)
- [x] [Android doesn't build due to missing `cliPath`](https://github.com/facebook/react-native/issues/28466)
- [x] [Android doesn't build due to incorrect Kotlin version](https://github.com/facebook/react-native/pull/28626)
</details>

## Non-Blocking

<details>
  <summary>Click to expand!</summary>

- [x] Bump Flipper to v0.41
- [x] Documentation for Pressable @kikisaints / @rachelnabors
- [x] Rename Pressable `pressRectOffset` prop to `pressRetentionOffset`. @TheSavior 
- [x] [Make sure PlatformColor ESLint rule is enabled on new projects](https://github.com/facebook/react-native/commit/602070f44b02220aeb036a7b3c26dad5c611b636) (might require publishing a new package)
- [x] [Remove ColorAndroid method](https://github.com/facebook/react-native-website/pull/1813#issuecomment-609962877) @tom-un
- [x] [Documentation for PlatformColor](https://github.com/facebook/react-native-website/pull/1813) @kikisaints / @rachelnabors
- [x] [Documentation for DynamicColorIOS](https://github.com/facebook/react-native-website/pull/1813) @kikisaints / @rachelnabors
- [x] [Fix Pressable's Android ripple with `borderless`](https://github.com/facebook/react-native/pull/28526) @vonovak. Fix [here](https://github.com/facebook/react-native/commit/44ec762e41029bf43530b1ff9b36ca3512c526e2)
- [x] [Fix Podfile paths to React Native in monorepos and non-standard structures](https://github.com/facebook/react-native/pull/28572) @grabbou
- [x] Ensure Hermes is on 0.5.0 @willholen (https://github.com/facebook/react-native/commit/4305a291a94)
- [x] Cherry-pick [Podfile cleanup](https://github.com/facebook/react-native/pull/28651#issuecomment-614663967)
- [x] Allow iOS PlatformColor strings to be ObjC or Swift UIColor selectors https://github.com/facebook/react-native/commit/25793eab56217a9961620761ea65ec2fcb97dcb0
- [x] facebook/react-native@d06ee3d
</details>

## Hopeful Dates

We can never commit to the dates involved in a release as many things are out of our control. However, we are thinking about these rough dates for this release.

- **Publish as Stable:** June

**Comment by HoSheiMa** [comment]
Fix android shadow system and make it use same props like ios system to be easy, please fix a very bad react dev debag it's make animation very slow when it opened, make support for react native to can use flooting windows

**Comment by ltcaosj** [comment]
Please include this into the release a well, there are so many people need it.
Allow touches on clipped children when `clipChildren==false` https://github.com/facebook/react-native/pull/26374

**Comment by grabbou** [comment]
During the testing, we've discovered a potential DX downgrade and submitted a PR to quickly resolve it: https://github.com/facebook/react-native/pull/28572.

Should be merged today - don't think it's a blocker.

**Comment by ralph-dev** [comment]
Just curious what the status is with **FlatList Versus SectionList** versus indirectly overlaps like ScrollView / Fast List. styled-components don't support .SectionList but does support .FlatList and ReactNative Web does not recommend either and pushes you towards Fast List. So from an integration perspective, I feel like at the minimum we should deprecate SectionList and improve FlatList significantly? 

This is I'm sure out of scope for the next release but I would like to hear something about this topic in general (At least a discussion to recognize the issue / Updating Docs if I am incorrect)

**Comment by TheSavior** [comment]
SectionList and FlatList are different and are both necessary. SectionList renders sections of list items with their own headers, FlatList renders a single flat list of items.

**Comment by chetstone** [comment]
Just curious-- what is the rationale for dropping support for iOS 9? I know it's old, but the form factor (fits nicely in the palm) of older devices that only run iOS 9.3.6 is ideal for my app...
If you can point me to a discussion that would be great..

Thanks.

**Comment by TheSavior** [comment]
@chetstone We periodically drop old iOS versions that we see very little usage of in order to be able to use newer features and ease development. 

A bit more concretely, as React Native is developed in the same monorepo as the Facebook app and all the other apps created by Facebook, we have to maintain the same support as them. The apps created by Facebook dropped support for iOS 9 at the end of last year.

**Comment by chetstone** [comment]
Thanks for the reply.

**Comment by grabbou** [comment]
Hit an Android error while running a new project: https://github.com/facebook/react-native/issues/28466

Investigating.

**Comment by grabbou** [comment]
I have submitted a PR to fix that issue: https://github.com/facebook/react-native/pull/28625. I am still working on it, it may change a bit before it's fully ready.

**Comment by grabbou** [comment]
Another issue on Android while building a project: 

> * What went wrong:
The Android Gradle plugin supports only Kotlin Gradle plugin version 1.3.10 and higher.
The following dependencies do not satisfy the required version:
root project 'RNTestProject' -> org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.0

Cause: https://github.com/facebook/react-native/blob/29d3dfbd196176af98c9727c82ff2668e697d78e/template/android/build.gradle

Fix: Set to 1.3.10.

**Comment by grabbou** [comment]
I have submitted a PR to fix that issue: https://github.com/facebook/react-native/pull/28626.

**Comment by KingAmo** [comment]
https://github.com/facebook/react-native/issues/21801 [Android][Animation] Interpolated translateY value causes android view to jump

@TheSavior could this issue be fix in `0.63`? it is really a big problem for all react native developer

**Comment by TheSavior** [comment]
@grabbou, version `@react-native-community/eslint-plugin@1.1.0` has been published https://www.npmjs.com/package/@react-native-community/eslint-plugin/v/1.1.0

**Comment by TheSavior** [comment]
@KingAmo only commits landed in master will be included in 63. That issue doesn't have a PR that has been landed

**Comment by grabbou** [comment]
Thanks! I have successfully tested Android and iOS projects and everything works just fine for me. We're ready to go.

**Comment by grabbou** [comment]
0.63.0-rc.0 is on the way - https://github.com/facebook/react-native/releases/tag/v0.63.0-rc.0. If everything goes well on CircleCI, we should have a new RC tonight.

**Comment by Simek** [comment]
I think it's a bit confusing that `PlatformColor` strings on iOS have additional `Color` suffix in comparison to the Apple docs:
* https://developer.apple.com/documentation/uikit/uicolor/ui_element_colors

Example: `secondarySystemFill` in RN is `secondarySystemFillColor`.

**Comment by tom-un** [comment]
Regarding the `Color` suffix on semantic names: in the Objective C APIs, the names *do* have a `Color` suffix, but in Swift they do not.   The iOS (and macOS) implementations are ObjC and are using ObjC reflection to validate and call the selector names -- and the selector names *do* have the `Color` suffix.   Its an interesting difference between the ObjC api and the Swift api that actually never noticed before.

**Comment by Simek** [comment]
@tom-un Thank you for the explanation. I have updated the reference link in the docs PR: facebook/react-native-website#1851.

**Comment by owinter86** [comment]
I am not sure if there needs to be a change to the local environment for 062.2 and 0.63 Release candidate, but I can not successfully build android release versions the new release candidate, but I have no problem when initialising a new 0.62.2 version.

https://github.com/facebook/react-native/releases

**Comment by TheSavior** [comment]
Thoughts from looking through the [upgrade helper from 62.2 -> 63.0-rc.0](https://react-native-community.github.io/upgrade-helper/?from=0.62.2&to=0.63.0-rc.0)

- package.json has eslint-config@1.0.0. I think this needs to get bumped to 1.1.0 to include the new ESLint rule
- I'm so happy with the Podfile cleanup

**Comment by tom-un** [comment]
@Simek and @TheSavior : on the iOS color selector names.   I just did a deep dive into Swift and back with the help of a Swift expert on our team.    Swift is the future for Apple.   They're docs and examples always default to Swift now.   It would be relatively easy to make iOS PlatformColor accept Swift selectors *and* ObjC selectors.   i.e. we could make `PlatformColor('labelColor')` and `PlatformColor('label')` both be legal.   Thoughts?

**Comment by Simek** [comment]
@tom-un If there are no potential issues with that approach I think that having both versions accepted would be an ideal solution.

**Comment by TheSavior** [comment]
I think supporting both is reasonable

**Comment by tom-un** [comment]
@Simek , @TheSavior and @alloy :  I've made a change to support both ObjC and Swift UIColor selector names here: https://github.com/facebook/react-native/compare/master...tom-un:tomun/platformcolor-swift.   I will make it into a PR.

**Comment by willholen** [comment]
When attaching a debugger to v0.63.0-rc.0 with Hermes, the app instantly crashes. Currently investigating.

**Comment by TheSavior** [comment]
Let's pick @tom-un's commit adding the other color names: https://github.com/facebook/react-native/commit/25793eab56217a9961620761ea65ec2fcb97dcb0

**Comment by alloy** [comment]
> I have updated the reference link in the docs PR: [facebook/react-native-website#1851](https://github.com/facebook/react-native-website/pull/1851).

@Esemesek I left a comment about perhaps reverting the link to the HIG instead, now that @tom-un’s change is in. https://github.com/facebook/react-native-website/pull/1851/files#r412254553

**Comment by grabbou** [comment]
Not updating to Flipper 0.37 yet - waiting for a commit from `master` to cherry-pick. However, Podfile cleanup works w/o the bump.

All the commits have been cherry-picked.

**Comment by grabbou** [comment]
I have also updated the original issue to hide already resolved commits.

**Comment by Simek** [comment]
@alloy Thank you for pointing this out. There is already a PR with this change: facebook/react-native-website#1865.

**Comment by grabbou** [comment]
> package.json has eslint-config@1.0.0. I think this needs to get bumped to 1.1.0 to include the new ESLint rule

@TheSavior, according to https://semver.npmjs.com/, `^1.0.0` will pick `1.1.0` too. Since there's no lock file shipped with the default template, I don't think it should have any impact.

On that note, it might be a good idea to include bumping the template dependency to be part of the ESLint config release process.

**Comment by grabbou** [comment]
@owinter86, this looks like a regression. We're investigating this. Thank you for highlighting.

**Comment by TheSavior** [comment]
@Grabbou 
> according to https://semver.npmjs.com/, ^1.0.0 will pick 1.1.0 too.

Yeah, people with existing projects and lock files won’t update though. We need everyone on 1.1.0 with this release so that they have the new lint rule :-)

**Comment by willholen** [comment]
Please pick https://github.com/facebook/react-native/commit/d06ee3d18973e74983590d2888552dbd57ceb951 

With this commit, you can attach and debug with Hermes. Without it, the app crashes instantly when attaching a debugger (due to a bad build setting).

**Comment by PierreCapo** [comment]
Hello,
Can this commit be cherry-picked https://github.com/facebook/react-native/commit/ed29ba13f97f240c91fdf6c0ef3fb601046697b9 please @grabbou  ?

It adds a nice feature to Android ScrollView and make it more consistent with the iOS ones

**Comment by TheSavior** [comment]
@PierreCapo, that looks like a long standing issue so we don't normally pick those into RCs. That will be part of 0.64

**Comment by grabbou** [comment]
Fix for @owinter86 issue is here: https://github.com/facebook/react-native/pull/28776.

**Comment by TheSavior** [comment]
@grabbou, that has landed on master in https://github.com/facebook/react-native/commit/afdcdc760f560963fdb48e409470d4119763d10b

**Comment by grabbou** [comment]
Yes, thanks @TheSavior! Picked and testing.

FYI I have a little problem with https://github.com/facebook/react-native/commit/17f025bc26da13da795845a3f7daee65563420c0 which results in:

```
[!] An error occurred while processing the post-install hook of the Podfile.

[!] The plist file at path `/Users/grabbou/Repositories/react-native/RNTester/RNTester.xcodeproj/project.pbxproj` doesn't exist.
```

produced by this line: https://github.com/facebook/react-native/commit/17f025bc26da13da795845a3f7daee65563420c0#diff-daf3e7bfe0b99f174d54291883fd60e3R100.

Probably needs to be investigated for next RC. Skipping it for now.

**Comment by grabbou** [comment]
I am also picking this https://github.com/facebook/react-native/commit/fb04237b18e1467889c865f3feb75ec85e807dd8#diff-daf3e7bfe0b99f174d54291883fd60e3 which is revert of a previously picked commit, which breaks the builds too.

**Comment by Naturalclar** [comment]
Can https://github.com/facebook/react-native/pull/28662 be included in v0.63-rc? so that initial template made by cli will include the latest eslint-config

**Comment by alloy** [comment]
> FYI I have a little problem with [facebook/react-native@17f025b](https://github.com/facebook/react-native/commit/17f025bc26da13da795845a3f7daee65563420c0) which results in:
> 
> ```
> [!] An error occurred while processing the post-install hook of the Podfile.
> 
> [!] The plist file at path `/Users/grabbou/Repositories/react-native/RNTester/RNTester.xcodeproj/project.pbxproj` doesn't exist.
> ```
> 
> produced by this line: [facebook/react-native@17f025b#diff-daf3e7bfe0b99f174d54291883fd60e3R100](https://github.com/facebook/react-native/commit/17f025bc26da13da795845a3f7daee65563420c0#diff-daf3e7bfe0b99f174d54291883fd60e3R100).

That entire `post_install` hook should not be there, we had removed it in 0.62. I’ll reply on https://github.com/facebook/react-native/commit/17f025bc26da13da795845a3f7daee65563420c0#diff-daf3e7bfe0b99f174d54291883fd60e3

**Comment by alloy** [comment]
> FYI I have a little problem with [facebook/react-native@17f025b](https://github.com/facebook/react-native/commit/17f025bc26da13da795845a3f7daee65563420c0) which results in:
> 
> ```
> [!] An error occurred while processing the post-install hook of the Podfile.
> 
> [!] The plist file at path `/Users/grabbou/Repositories/react-native/RNTester/RNTester.xcodeproj/project.pbxproj` doesn't exist.
> ```
> 
> produced by this line: [facebook/react-native@17f025b#diff-daf3e7bfe0b99f174d54291883fd60e3R100](https://github.com/facebook/react-native/commit/17f025bc26da13da795845a3f7daee65563420c0#diff-daf3e7bfe0b99f174d54291883fd60e3R100).

@grabbou @TheSavior I’ve created 2 PRs to address this once and for all, however it does depend on a new Flipper version being release before we can pull it into the RC:

* https://github.com/facebook/flipper/pull/1086
* https://github.com/facebook/react-native/pull/28796

**Comment by TheSavior** [comment]
> Can facebook/react-native#28662 be included in v0.63-rc? so that initial template made by cli will include the latest eslint-config

This landed and should get picked. https://github.com/facebook/react-native/commit/780f06cd477f34da48646a949bd25dd3f883a9a2

**Comment by grabbou** [comment]
Picked @TheSavior 👌

**Comment by grabbou** [comment]
0.63.0-rc.1 is on CircleCI with latest round of cherry-picks. Should be on `npm` soon.

**Comment by danielgindi** [comment]
How about finally merging #26374 to get rid of that damn Android issue and make it comparable to iOS?

**Comment by alloy** [comment]
Flipper 0.41.1 with simplified CocoaPods/Xcode setup https://github.com/facebook/react-native/commit/7bb1c4e1b8715a5c9cb6f9e4e77a6df783481d3d.

**Comment by TheSavior** [comment]
@danielgindi, as that PR hasn't landed, it won't be a part of 0.63.

**Comment by gaodeng** [comment]
react native 0.63.0-rc.1 enableHermes: true, As soon as bluebird is imported, an unknown error occurs  https://github.com/facebook/hermes/issues/241

**Comment by danielgindi** [comment]
@gaodeng bluebird should not be used anymore, as `Promise` is native to any modern ES engine. on the other hand, hermes is a crippled engine, and probably does not follow the spec, so bluebird's `Promise`-like objects probably confuses it. You could switch to `v8` engine, at least for Android, to get better performance. I got no idea why are they promoting hermes.

**Comment by trevyn** [comment]
@danielgindi My understanding is that Hermes is intended to reduce app startup time and memory usage, not to be spec-compliant. If you don't need that, don't use it I suppose.

**Comment by TheSavior** [comment]
@danielgindi, as far as is known, Hermes is spec compliant on the features that it implements. It verifies against the same spec test suite as other engines. Any non compliance is a bug that should be reported to the Hermes repo.

As for Hermes being slower than V8, these are the results of some experiments Microsoft ran comparing V8 against Hermes on one of their production apps. On all of these metrics Hermes outperforms v8. 

![66255648_893501134346100_8745316183667900416_n](https://user-images.githubusercontent.com/249164/81372978-877d4880-90b0-11ea-8784-4ff5e6eb1522.png)

Here is a comment from @acoates-ms on that topic who works on Office: https://news.ycombinator.com/item?id=20413046

Please consider avoid spreading misinformation

**Comment by danielgindi** [comment]
> Hermes is spec compliant on the features that it implements. It verifies against the same spec test suite as other engines. Any non compliance is a bug that should be reported to the Hermes repo.
Then there might be a bug with Promises there. Also when Hermes was first paired with RN, there was a lot of talk about so many ES features not supported, so babel is needed to bloat everything.

Also - how is that size reasonable if it's spec compliant, and let's say - supports `Intl`?.. Apps these days need `Intl` as everything is global now.


> 
> As for Hermes being slower than V8, these are the results of some experiments Microsoft ran comparing V8 against Hermes on one of their production apps. On all of these metrics Hermes outperforms v8.

Have you seen these benchmarks?
https://github.com/Kudo/react-native-js-benchmark
https://docs.google.com/spreadsheets/d/1uce3WZ9IaAEUu6Owj3eXEuZb25PDi6ZcgUVV2i500S0/edit#gid=1258377944

It's not only about having a smaller footprint or 10ms less in boot. It's about the runtime itself that should be faster and consume less memory.

**Comment by TheSavior** [comment]
This isn't the place for this conversation. This thread is specifically focused on the release process for 0.63. If you have additional questions about Hermes and the things it supports, an issue on the Hermes repo is a good choice.

**Comment by TheSavior** [comment]
@rachelnabors noticed that the Pressable prop `pressRectOffset` isn't consistent with other touchables. I'm making a change to rename that prop and we should pick that change once it lands.

**Comment by fkgozali** [comment]
Might want to pick https://github.com/facebook/react-native/commit/e5a6655e71d41a58ce0e51d37aa9fb8792e37dd5, that should fix long-standing iOS crashes: https://github.com/facebook/react-native/issues/27463

**Comment by TheSavior** [comment]
I think we should pick https://github.com/facebook/react-native/commit/8ac467c51b94c82d81930b4802b2978c85539925 as well. We forgot to make Pressable have the same debug helper as the existing Touchables.

**Comment by rachelnabors** [comment]
Docs are ready to be cut. We have `Pressable` and `DynamicColors` ready to go!

**Comment by pmachowski** [comment]
For me after upgrading to the latest release (0.63.0-rc.1) with `Podfile` cleanup approach resulted in adding user specific, absolute paths to RN modules in `Podfile.lock` instead of relative ones. 

Does anyone experienced this behavior?

So instead of
```
React-Core (from `../node_modules/react-native/`)
```
I get
```
React-Core (from `/Users/pmachowski/Work/Mobile/Project/node_modules/react-native/`)
```

**Comment by grabbou** [comment]
Just heads up - I am back from holidays and looking into the state of the release. I will be back with updates.

**Comment by grabbou** [comment]
All commits cherry-picked. Working on the release now.

**Comment by pvinis** [comment]
@pmachowski would you mind telling us how did you upgrade, and where did you see these paths? thank you.

**Comment by pmachowski** [comment]
@pvinis Sure. I've upgraded using the React Native Upgrade Helper. But you can reproduce it with clean install: `npx react-native init ProjectName --version 0.63.0-rc.1`. Absolute paths are located in `ProjectName/ios/Podfile.lock`

**Comment by danielgindi** [comment]
@pmachowski I think that cocoapods version is relevant here

**Comment by lukebars** [comment]
@danielgindi I reproduced absolute paths in `Podfile.lock` with `COCOAPODS: 1.9.1`

**Comment by grabbou** [comment]
We're working on the changelog/blog post now, the release is ready from the code perspective. The ETA is next Monday/Tuesday.

**Comment by alloy** [comment]
@grabbou With a new v0.63.0-rc.1 application and using CocoaPods v1.8.4 I can reproduce the same absolute path issue. Did the way we provide the paths to the podspecs change?

**Comment by pmachowski** [comment]
@danielgindi @pvinis my bad. I should have provide it. Like others has already reported the version of Cocoapods affected is `1.9.2`

**Comment by alexwasner** [comment]
Looks like the release is right around the corner, but I upgraded to from 62.2 to rc1 and all my tests are failing. 
```
 FAIL  __tests__/Menu.test.tsx
  ● Test suite failed to run
    TypeError: Cannot convert undefined or null to object
        at assign (<anonymous>)
      at _extends (node_modules/@babel/runtime/helpers/extends.js:16:19)
      at node_modules/@testing-library/react-native/dist/preset/mock-modules.js:41:25
      at Object.require (node_modules/react-native/Libraries/Image/Image.ios.js:15:21)
```

I have upgraded all of the jest and babel versions from [the helper](https://react-native-community.github.io/upgrade-helper/). Any insight into the specifics around the jest updates?

**Comment by TheSavior** [comment]
@alexwasner, the changelog (in progress) is here: https://github.com/react-native-community/releases/pull/198 You can see the changes from Jest in there. There isn't enough information here to debug your problem so it is hard to tell. I'd recommend trying to create repro in a new project on 0.63.0-rc1 to narrow it down and create a new issue on the main React Native repo with that info.

**Comment by codecog** [comment]
With the cleanup of the podfile in 0.63.0, how would one pass an updated version number through to `use_flipper!`? I think currently RN uses 0.33 but flipper is at 0.44.1 right now.

**Comment by kelset** [comment]
@codecog I'm not sure if that's the case, but my guess would be that it's the same thing that happens with the React and Flow versions: they are "bound" to that version of React Native, so you shouldn't upgrade to latest unless RN explicitly supports it.

That said, probably cocoapods has some resolution mechanism if you were to define the Pod for flipper anyway in your podfile? (again, not sure)

**Comment by alexwasner** [comment]
Per my testing issue, I've narrowed it down to `@testing-library/react-native` and have replaced it with `react-native-testing-library` for now. For any future dev reading this, I've [opened a ticket](https://github.com/testing-library/native-testing-library/issues/126) with them with an [example repo](https://github.com/alexwasner/react-native-63-jest-bug) if you'd like to follow along.

**Comment by dulmandakh** [comment]
pod install with **use_frameworks!** failing, and IMHO, it's very important to support use_frameworks!. Filed an issue https://github.com/facebook/react-native/issues/29051

**Comment by nicoburns** [comment]
> With the cleanup of the podfile in 0.63.0, how would one pass an updated version number through to use_flipper!? I think currently RN uses 0.33 but flipper is at 0.44.1 right now.

A related question: we currently have multiple "debug" configurations (one for our production environment, one for our dev environment). We had to alter our Podfile like so:

```ruby
project 'marketplace',
  'Dev.Debug' => :debug,
  'Dev.Release' => :release,
  'Prod.Debug' => :debug,
  'Prod.Release' => :release

def add_flipper_pods!(versions = {})
  versions['Flipper'] ||= '~> 0.33.1'
  versions['DoubleConversion'] ||= '1.1.7'
  versions['Flipper-Folly'] ||= '~> 2.1'
  versions['Flipper-Glog'] ||= '0.3.6'
  versions['Flipper-PeerTalk'] ||= '~> 0.0.4'
  versions['Flipper-RSocket'] ||= '~> 1.0'
  pod 'FlipperKit', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/FlipperKitLayoutPlugin', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/SKIOSNetworkPlugin', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/FlipperKitUserDefaultsPlugin', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/FlipperKitReactPlugin', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  # List all transitive dependencies for FlipperKit pods
  # to avoid them being linked in Release builds
  pod 'Flipper', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'Flipper-DoubleConversion', versions['DoubleConversion'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'Flipper-Folly', versions['Flipper-Folly'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'Flipper-Glog', versions['Flipper-Glog'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'Flipper-PeerTalk', versions['Flipper-PeerTalk'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'Flipper-RSocket', versions['Flipper-RSocket'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/Core', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/CppBridge', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/FBCxxFollyDynamicConvert', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/FBDefines', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/FKPortForwarding', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/FlipperKitHighlightOverlay', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/FlipperKitLayoutTextSearchable', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
  pod 'FlipperKit/FlipperKitNetworkPlugin', versions['Flipper'], :configurations => ['Dev.Debug', 'Prod.Debug']
end
# Post Install processing for Flipper
def flipper_post_install(installer)
  installer.pods_project.targets.each do |target|
    if target.name == 'YogaKit'
      target.build_configurations.each do |config|
        config.build_settings['SWIFT_VERSION'] = '4.1'
      end
    end
  end
end
```

Is there a way to set up custom configurations like this with `use_flipper!`?

**Comment by alloy** [comment]
@nicoburns Adding an additional parameter to `use_flipper!` that takes a list of configurations and defaults to just `Debug` seems like a good change to me 👍 Please send a PR for that 🙏

**Comment by nicoburns** [comment]
@alloy PR here https://github.com/facebook/react-native/pull/29074

**Comment by DmitryAvramenko** [comment]
Almost two weeks on the changelog. Is this a joke?

**Comment by TheSavior** [comment]
@alloy Is the issue in the pod lockfiles with absolute paths a launch blocker?

**Comment by grabbou** [comment]
@TheSavior, my understanding is that this is not an issue as it affects paths in a lockfile generated on a post-install. But I'd love confirmation from @alloy just to make sure I am not getting something wrong.

**Comment by lukebars** [comment]
@grabbou Wouldn't it result in podfile.lock changes every time someone else installs pods even if Pods aren't changed? I would imagine that being quite an issue for PR's and commits on bigger projects.

**Comment by TheSavior** [comment]
Yeah, it seems like you wouldn’t be able to check in a lock file for a team for example

**Comment by Ashoat** [comment]
Wanted to chime in that I do believe this issue should be launch-blocking. `cocoapods` recommends checking in the `Podfile.lock`, similar to how `yarn.lock` works.

**Comment by codecog** [comment]
Is there anything on v0.63.0 that I could potentially help with to get it closer to release?

**Comment by kesha-antonov** [comment]
How to use relative paths in Podfile:

```
use_native_modules!
use_react_native!(path: '../node_modules/react-native')
```

I hope this helps to push the release

**Comment by grabbou** [comment]
@kesha-antonov this will be a breaking/backward-step for those that rely on mono repositories or non-standard locations (where it's not `../node_modules/react-native`). Let me check whether this is the root cause (`config["reactNativePath"]` is absolute) and if yes, change it to a relative path.

**Comment by grabbou** [comment]
**Update:** Together with the React Native team at Facebook, we have decided that last week was not the right timing for a stable React Native release. We want to step back and make space for more important voices to be heard. 

I will update this issue when the new date is set at a more appropriate time.

In the meantime, we encourage you to visit https://blacklivesmatter.com/ and learn more about how you can help.

**Comment by reimertz** [comment]
Thanks @grabbou for the hard work and agreed regarding the timing. <3

**Comment by NelsonPRSousa** [comment]
> Can we cherry-pick [facebook/react-native#29072](https://github.com/facebook/react-native/pull/29072)?
> 
> This fixes a security issue on a dependency from react-native-cli.

I am using 0.62.2 and got rid of that security issue by deleting yarn.lock and run `yarn` again. 0.7.1 seems to be compatibly and it's working as expected.

**Comment by Gustash** [comment]
> > Can we cherry-pick [facebook/react-native#29072](https://github.com/facebook/react-native/pull/29072)?
> > 
> > This fixes a security issue on a dependency from react-native-cli.
> 
> I am using 0.62.2 and got rid of that security issue by deleting yarn.lock and run `yarn` again. 0.7.1 seems to be compatible and it's working as expected.

Yeah mate, I'm sorry but deleting a lock file really isn't an ideal solution. There's a reason they exist

**Comment by Bardiamist** [comment]
@Gustash You can delete `yarn.lock` only once. Then `yarn` will create new `yarn.lock` with latest `@react-native-community/cli` (`4.10.0`)

**Comment by Gustash** [comment]
> @Gustash You can delete `yarn.lock` only once. Then `yarn` will create new `yarn.lock` with latest `@react-native-community/cli` (`4.10.0`)

The problem isn't that the lock file won't regenerate, it's that it'll literally bring the latest version of all your dependencies, and you can't trust all of them to not have breaking changes. 

Therefore, for big projects, deleting a lock file is a big no-no

**Comment by kelset** [comment]
@Gustash while in line of principle, yes, you shouldn't remove a yarn.lock, there are actually 2 different work arounds that won't require the cherry pick (that said I think it should be cherry picked).

1. go in the yarn lock and delete manually only the entries around rn-cli (the 4 packages indicated above), then do a yarn and it will pull down the latest 4.x

1. use [yarn resolutions](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/#toc-how-to-use-it) to set the RN-cli dependencies to the desired version.

**Comment by Gustash** [comment]
@kelset do those versions of the cli work properly with 0.62.2? I'm weary of using cli versions not set by RN because I've had build issues before

**Comment by kelset** [comment]
@Gustash since the commit you propose to cherry pick is from 4.7 to 4.10 I'd say it's safe to "manually upgrade" those packs on the side anyway. The [release notes](https://github.com/react-native-community/cli/releases) don't indicate any breaking changes / minimum version either.

**Comment by Gustash** [comment]
Thanks @kelset, that's a much more sensible solution. 

I would still argue for cherry-picking this though

**Comment by tad3j** [comment]
On windows build is failing with:
```
'..\\..\\..\\..\\node_modules\\hermes-engine\\win64-bin\\hermes' is not recognized as an internal or external command,
```
This seems to work with RN 0.62.2.

I use hermesCommand to provide the location of hermes like:
```
project.ext.react = [
  entryFile   : "index.js",
  enableHermes: true,  // clean and rebuild if changing
  hermesCommand: "../../../../node_modules/hermes-engine/%OS-BIN%/hermes",
]
```

**Comment by grabbou** [comment]
Fix for the `Podfile.lock` paths issue (based on @kesha-antonov workaround) is here: https://github.com/react-native-community/cli/pull/1203

CC: @alloy

**Comment by emilisb** [comment]
Can we cherry-pick this? https://github.com/facebook/react-native/pull/27885

It should have no negative impact. It offers new benchmarking opportunities.

**Comment by piashcse** [comment]
@grabbou So when will we get the 0.63 stable?

**Comment by grabbou** [comment]
@Gustash I have not cherry-picked this commit, but I have bumped CLI to even newer version, should be fixed! Along with that, the Podfile fix is also there.

@piashcse, as per the previous comment, I will let you know as soon as such date is set.

@emilisb, your request is not a critical bug fix which would qualify it as a good candidate to be cherry-picked. However, I understand the importance of it for you and have passed it for discussions internally. Will be back with the updates really soon.

**Comment by owcc** [comment]
_"We want to step back and make space for more important voices to be heard"_

Can I understand that the delay of v63.0 release may because support for BLM? If so, RN team is so political.

**Comment by Gustash** [comment]
@owcc there is nothing political about human rights.

**Comment by gaodeng** [comment]
> _"We want to step back and make space for more important voices to be heard"_
> 
> Can I understand that the delay of v63.0 release may because support for BLM? If so, RN team is so political.

This is just a bad excuse for procrastination. The last version release process took almost half a year

**Comment by mikehardy** [comment]
@gaodeng how'd you like that open source repository where you could just go ahead and take advantage of the code without waiting on anyone? Pretty cool right? Your cynicism is sad and contributions are zero https://github.com/react-native-community/releases/pulls/gaodeng, so...

**Comment by mikehardy** [comment]
Let's keep this on topic people. If you're looking for cherry picks for the next release, and it has a commit hash, post them here.

**Comment by gaodeng** [comment]
> @gaodeng how'd you like that open source repository where you could just go ahead and take advantage of the code without waiting on anyone? Pretty cool right? Your cynicism is sad and contributions are zero https://github.com/react-native-community/releases/pulls/gaodeng, so...

so  https://github.com/react-native-community/releases/pulls/mikehardy  
 If the RN release process is really so difficult, then we should learn how competing products flutter are released regularly

**Comment by owcc** [comment]
> @owcc there is nothing political about human rights.

It's true, but with wisdom and civilization. I just feel sad for what happened in past days in US, robbery, riot and killings, sometimes we may not see the truth, but life has to continue. I live in a worse country but I still believe god love everyone. Let me end this topic, I just want leave technology be pure and continue

**Comment by farshed** [comment]
I wholeheartedly support the BLM movement but I don't understand what's it got to do with React Native or how not making a release is supposed to help BLM. Is this some kind of protest?

**Comment by owcc** [comment]
> I wholeheartedly support the BLM movement but I don't understand what's it got to do with React Native or how not making a release is supposed to help BLM. Is this some kind of protest?

“make space for more important voices to be heard” what is this "voices"? I'm sorry for a misunderstanding if it means "more tests or less bugs or constructive opinion for this release", but if it means "let's make some time to learn what's BLM and may join us to support ", that's the way "how not making a release is supposed to help BLM". Because "the BLM topic" is appeared [here](https://github.com/react-native-community/releases/issues/186#issuecomment-641108746) make me had to think the connection  about this.

Again, I appreciate the great work of RN team.

**Comment by cpojer** [comment]
Locking this issue to maintainers as the discussion went off topic and wasn't focused on the actual release any longer. We'll release 0.63.0 soon.

**Comment by grabbou** [comment]
As per this https://github.com/react-native-community/releases/issues/199#issuecomment-652284341 comment, we have cherry-picked https://github.com/facebook/react-native/commit/33b3a1a1453ca51690e59b758eeb61a4fa8f35bc for compatibility with latest Apple requirements.

**Comment by grabbou** [comment]
Changelog PR is ready and can be merged anytime the release is out.

**Comment by grabbou** [comment]
The blog post PR is ready and can be merge anytime the release is out.

**Comment by grabbou** [comment]
Before we finally push the release out, we need to make sure that https://github.com/facebook/react-native/pull/29204 is merged and cherry-picked to fix issues on Android.

**Comment by grabbou** [comment]
The fix has landed in 8c42c01. I have already cherry-picked it. Because it's a bit late, I will postpone the final round of testing and the release till tomorrow. Meanwhile, I will let the CI process the changes and ensure the tests are still passing.

**Comment by sunnylqm** [comment]
@grabbou What about this issue? https://github.com/facebook/react-native/issues/28551  The fix is to update flipper https://github.com/facebook/react-native/pull/29033

**Comment by alloy** [comment]
@sunnylqm While a good fix, it has not landed in master yet and does not appear to be a blocker for this release—as it only applies to some and can be worked around, whereas _nobody_ being able to debug on Android is a deal-breaker. Once your fix lands, please ask it to be picked on [the v0.63.1 issue](https://github.com/react-native-community/releases/issues/199) instead.

**Comment by grabbou** [comment]
0.63.0 is being released as we speak, closing this issue in the meantime. Let's move on to the 0.63.1 from now on.
